### DISCUSSION TITLE
Creating data for segmentation task

### DISCUSSION THREAD

**Discussion by melikycode** [ultralytics/ultralytics#6389] [Q&A]
Hi,

I have a data of segmentation masks in a form of images. Since Yolov8 takes polygon coordinates, I am manually creating them. I am stuck at an issue. Most of my mask blobs have holes in it like following image.

![Screenshot 2023-11-16 at 7 19 34 PM](https://github.com/ultralytics/ultralytics/assets/117266901/d6ee9e75-b9b0-4fcf-b176-02e9ada8ae51)

I do have polygon points for both outer and inner blob but is there is a way represent that in the data ? I would really appreciate your help.

**Comment by glenn-jocher** [comment]
@melikycode hey thanks for asking! Right now separate regions in the YOLO segmentation format can be connected with very thin lines, so i.e. the outer and inner outlines form a single line and are bridged by a zero-width line. Does this make sense?
